### PR TITLE
Jetpack: Remove 'new site' test

### PR DESCRIPTION
**Pull request by johnHackworth**
Well, not a big story to tell here: 

The type of new site interstitial test have been deemed successful, so we are removing the test and opening this up for 100% of the users, yay.

How to test
=======

1. Go to calypso
2. "switch site" and the "add a new site"
3. You should see 
image

### CODE REVIEW

File: client/components/site-selector/add-site.jsx [hunk 1]
```diff
@@ -36,11 +34,7 @@ class SiteSelectorAddSite extends Component {
 	}
 
 	getAddNewSiteUrl() {
-		if ( this.props.hasJetpackSites ||
```
Comment (tyxla) [inline client/components/site-selector/add-site.jsx]: Just to verify: previously we were displaying the new Jetpack page when the user has a Jetpack site or when running the AB test. This change will show the new "new site" page to all users, even to those who don't have a Jetpack site. Let's make sure that's intentional.

Comment (johnHackworth) [inline client/components/site-selector/add-site.jsx]: yup, it is :)

**Comment by johnHackworth** [comment]
@unDemian  damn, right, I didn't include that file in the commit, sorry! Fixed!

**Comment by lancewillett** [comment]
@johnHackworth Ready to merge?

**Comment by rachelmcr** [comment]
@johnHackworth Just checking in on the status of this PR; looks like it has some conflicts to resolve and then it can be merged?

**Comment by rickybanister** [comment]
reping @johnHackworth

**Comment by johnHackworth** [comment]
oh, damn it. All those pings came while I was AFK and this totally fell through the cracks. I was totally convinced that this was merged already!!! I just found the last ping in my email and saw this PR hanging here !!!

I would have sworn that this was merged already (In fact I think that I did swear it to some of you). I'm sorry about this kerfluffle. I'll merge it right away.

**Pull request merged**